frontend/src/modules/SimulationTimeSeriesMatrix/dataGenerators.ts

```diff
@@ -0,0 +1,55 @@
+import { VectorRealizationData_api } from "@api";
+import { DataGenerator, ModuleChannelContentMetaData } from "@framework/DataChannelTypes";
+import { EnsembleIdent } from "@framework/EnsembleIdent";
+import { simulationUnitReformat, simulationVectorDescription } from "@modules/_shared/reservoirSimulationStringUtils";
+
+import { indexOf } from "lodash";
+
+import { VectorSpec } from "./state";
+
+export function makeVectorGroupDataGenerator(
+    vectorSpecificationsAndRealizationData: {
+        vectorSpecification: VectorSpec;
+        data: VectorRealizationData_api[];
+    }[],
+    activeTimestampUtcMs: number,
+    makeEnsembleDisplayName: (ensembleIdent: EnsembleIdent) => string
+): (contentIdString: string) => ReturnType<DataGenerator> {
+    return (contentIdString: string) => {
+        const [vectorName, ensembleIdent] = contentIdString.split("-::-");
+        const data: { key: number; value: number }[] = [];
+        let metaData: ModuleChannelContentMetaData = {
+            unit: "",
+            ensembleIdentString: "",
+            displayString: "",
+        };
+
+        const vector = vectorSpecificationsAndRealizationData.find(
+            (vec) =>
+                vec.vectorSpecification.vectorName === vectorName &&
+                vec.vectorSpecification.ensembleIdent.toString() === ensembleIdent
+        );
+        if (vector) {
+            let unit = "";
+            vector.data.forEach((el) => {
+                unit = simulationUnitReformat(el.unit);
+                const indexOfTimestamp = indexOf(el.timestamps_utc_ms, activeTimestampUtcMs);
```

Comment:
Is there a specific reason to use `lodash.indexOf`-method rather than `el.time_stamps_utc_ms.indexOf(activeTImestampUtcMs)`?

Reply:
There is no real reason for using this one as both work exactly the same way. I did actually only use it due to it already having been used in `SimulationTimeSeries` and did not give it another thought as it works as a 1-to-1 replacement. However, I think you have a point by rather sticking to standard functions. Moreover, I ran a benchmark at the native one seem to work considerably faster. So I am going to change this all places.